js/DataService.js

```diff
@@ -8,13 +8,46 @@ angular.module('PathOfDamage')
     var int = parseInt(string);
     return isNaN(int) ? null : int;
   }
+  
+  function Row(key, value) {
+    this.enabled = true;
+    this.name = '';
+    this.value = null;
+    if (key !== undefined) {
```

Comment:
The reason the key/value pair is optional is for making the encoding easier later, right? That we don't have parameters that we aren't going to use attached to the generic row? That should probably be the responsibility of the encoder instead of shaping the data structure around it. I would be fine having an 'element' property for every row, and only using it where appropriate, since more than half of our distinct tables use them.